FinancialTracker/Firebase/DatabaseManager.swift

```diff
@@ -492,29 +508,149 @@ class DatabaseManager {
+        var myTransactionType: Category
+    
+        switch transfer.transferType {
+        case .receive:
+            otherUserTransactionType = ExpenseCategory.transfer
+            myTransactionType = IncomeCategory.transfer
+        case .requestToMe:
+            otherUserTransactionType = IncomeCategory.transfer
+            myTransactionType = ExpenseCategory.transfer
+        default:
+            assertionFailure("Should not appear.")
+            return
+        }
+        
+        self.addTransactionToUserByUID(transfer.fromUser, amount: transfer.amount, category: otherUserTransactionType) { databaseError, _ in
+            if let databaseError = databaseError {
+                completionHandler(databaseError, false)
+                return
+            }
+            
+            // swiftlint:disable:next line_length
+            self.firestore.collection(DBCollectionKey.users.rawValue).document(transfer.fromUser).collection(DBCollectionKey.transfers.rawValue).document(transfer.uid).updateData([Transfer.TransferKeys.transferState.rawValue: "Completed"])
+            
+            self.addTransactionToUserByUID(currentUser.uid, amount: myTransferAmount, category: myTransactionType) { databaseError, _ in
+                if let databaseError = databaseError {
+                    completionHandler(databaseError, false)
+                    return
+                }
+                
+                // swiftlint:disable:next line_length
+                self.firestore.collection(DBCollectionKey.users.rawValue).document(currentUser.uid).collection(DBCollectionKey.transfers.rawValue).document(transfer.uid).updateData([Transfer.TransferKeys.transferState.rawValue: "Completed"])
+                
+                completionHandler(nil, true)
+            }
+        }
+    }
+    
+    /// Listener for firestore changes
+    /// - Parameters:
+    ///   - completionHandler: Block that is to be executed if an error appears or the function is successfully executed.
+    ///     1. `databaseError` - An error object that indicates why the function failed, or nil if the was successful.
+    ///     2. `user` - The user that was returned by firestore.
+    func firestoreDidChangeUserTransfersData(completionHandler: @escaping (DatabaseError?, [Transfer]?) -> Void) {
+        guard let currentUser = currentUser else {
+            completionHandler(DatabaseError.access("User data is nil \(#function)."), nil)
+            return
+        }
+        
+        // swiftlint:disable:next line_length
+        firestore.collection(DBCollectionKey.users.rawValue).document(currentUser.uid).collection(DBCollectionKey.transfers.rawValue).addSnapshotListener { query, error in
+            
+            guard error == nil else {
+                completionHandler(DatabaseError.database(error), nil)
+                return
+            }
+            
+            guard let query = query else {
+                return
```

Comment:
handle missing `query`??

Reply:
same as above

Reply:
same as above

Reply:
Also, all times you handle the query, handle it in the same way please